x-pack/plugins/triggers_actions_ui/public/common/lib/kibana/use_spaces_data.tsx

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useState, useEffect } from 'react';
+import { SpacesData } from '@kbn/spaces-plugin/public';
+import { useKibana } from './kibana_react';
+
+export const useSpacesData = () => {
+  const { spaces } = useKibana().services;
+  const [spacesData, setSpacesData] = useState<SpacesData | undefined>(undefined);
+  const spacesService = spaces?.ui.useSpaces();
+
+  useEffect(() => {
+    (async () => {
+      const result = await spacesService?.spacesDataPromise;
```

Comment:
Im not too familiar with how this spacesService works, but is this a network request? I noticed we're calling this both in the cell renderer component and the list component. Perhaps we should just pass it in if it's expensive.

Reply:
ah, good to hear